src/system-module.adoc

```diff
@@ -0,0 +1,257 @@
+with the version of that module being determined by the spec-version of each segment.
+
+The specific system symbols are largely uninteresting to users since they don't affect the text
+content that they interact with.
+The system macros are more visible, especially to authors of macros.
+
+This chapter catalogs the system-provided macros.
+The examples below use unqualified names, which works assuming no other module exports the same
+name, but the unambiguous form `:$ion:__macro-name__` is always correct.
+
+
+IMPORTANT: This list is not complete. We expect it to grow and evolve as we gain experience
+writing macros.
+
+
+=== Primitive Operators
+
+This section describes operators that cannot be defined as macros.
+
+
+==== Stream Constructors
+
+===== `void`
+
+[{nrm}]
+----
+(void) \-> any?
+----
+
+Produces an empty stream.
+The most common use of this operator is to supply “no value” to a voidable parameter.
+To make such use more readable, the special-case E-expression `(:)` is synonymous to `(:void)`.
+
+
+===== `values`
+
+[{nrm}]
+----
+(values (v any\...)) \-> any*
+----
+
+Produces a stream from any number of arguments, concatenating the streams produced by the nested
+expressions.
+Used to aggregate multiple values or sub-streams to pass to a single argument, or to return
+multiple results. Generally only useful with more than one subexpression.
+
+
+==== Value Constructors
+
+===== `make_string`
+
+[{nrm}]
+----
+(make_string (content text\...)) \-> string
+----
+
+Produces a non-null, unannotated string containing the concatenated content produced by the
+arguments. Nulls and annotations are discarded.
+
+TODO Probably useful to allow some other Ion scalars (at least) to allow type conversion.
+I think this would be most useful for ints, since the binary representation is more compact than
+as characters. Lobs wouldn’t work well, though.
+
+
+===== `make_symbol`
+
+[{nrm}]
+----
+(make_symbol (content text\...)) \-> symbol
+----
+
+Like `make_string` but produces a symbol.
+
+
+===== `make_list`
+
+[{nrm}]
+----
+(make_list (vals any\...)) \-> list
+----
+
+Produces a non-null, unannotated list from any number of inputs.
+Template expressions of the form `[E~1~, …, E~n~]` are equivalent to `(make_list E~1~ … E~n~)`.
+
+
+===== `make_sexp`
+
+[{nrm}]
+----
+(make_sexp (vals any\...)) \-> sexp
+----
+
+Like make_list but produces a sexp.
+
+[{nrm}]
+----
+(:make_sexp)      ⇒ ()
+(:make_sexp null) ⇒ (null)
+----
+
+
+===== `make_struct`
+
+[{nrm}]
+----
+(make_struct (kv any\...)) \-> struct
+----
+
+Produces a non-null, unannotated struct from any number of elements.
+The ``kv``s are processed in order, incrementally adding fields to an initially-empty struct.
+Various forms of ``kv``s are allowed:
+
+* A (non-null) string or symbol is treated as a field name, and MUST be followed by another value
+to comprise a key-value pair in the result. Annotations on the field name are discarded.
```

Comment:
Just checking, do we ignore annotations in other places too (vs error)? e.g. in the arguments for `make_string` and `annotate`?

Reply:
It's a bit inconsistent. #254